serde_codegen/src/attr.rs

```diff
@@ -177,31 +177,29 @@ pub struct FieldAttrs {
     name: Name,
     skip_serializing_field: bool,
     skip_deserializing_field: bool,
-    skip_serializing_field_if: Option<P<ast::Expr>>,
+    skip_serializing_if: Option<ast::Path>,
```

Comment:
I see. Given that it's not on our roadmap (no issue for it) and I am not convinced that it would be a valuable thing to support, I do no think we should keep around a lot of convoluted dead code in the hope that some day it will be useful.